src/Element/Select2.php

```diff
@@ -202,10 +203,25 @@ public static function preRenderSelect($element) {
     $current_language = \Drupal::languageManager()->getCurrentLanguage();
     $current_theme = \Drupal::theme()->getActiveTheme()->getName();
     $select2_theme_exists = \Drupal::service('library.discovery')->getLibraryByName($current_theme, 'select2.theme');
+
+    // Placeholder should be taken from #placeholder property if it set.
+    // Otherwise we can take it from '#empty_option' property.
+    $placeholder = $required ? new TranslatableMarkup('- Select -') : new TranslatableMarkup('- None -');
+    $placeholder = ['id' => '', 'text' => $placeholder];
```

Comment:
Reassigning this variable is a bit odd, we could rename the placeholder variable in line 209 to placeholder_text, or something similar